### PR TITLE
Perform a hard exit if the framework encounters an exception during i…

### PR DESCRIPTION
**Pull request by verma7**
…nitialization

For example, we encounter this situation when the HTTP port that the framework listens to is already in use.

### CODE REVIEW

**Comment by verma7** [comment]
We see this stack trace:
```
ERROR [2017-01-03 23:01:39,517] io.dropwizard.cli.ServerCommand: Unable to start server, shutting down
! java.net.BindException: Address already in use
! at sun.nio.ch.Net.bind0(Native Method) ~[na:1.8.0_65]
! at sun.nio.ch.Net.bind(Net.java:433) ~[na:1.8.0_65]
! at sun.nio.ch.Net.bind(Net.java:425) ~[na:1.8.0_65]
! at sun.nio.ch.ServerSocketChannelImpl.bind(ServerSocketChannelImpl.java:223) ~[na:1.8.0_65]
! at sun.nio.ch.ServerSocketAdaptor.bind(ServerSocketAdaptor.java:74) ~[na:1.8.0_65]
! at org.eclipse.jetty.server.ServerConnector.open(ServerConnector.java:321) ~[jetty-server-9.2.14.v20151106.jar:9.2.14.v20151106]
! at org.eclipse.jetty.setuid.SetUIDListener.lifeCycleStarting(SetUIDListener.java:200) ~[jetty-setuid-java-1.0.3.jar:na]
! ... 8 common frames omitted
! Causing: java.lang.RuntimeException: java.net.BindException: Address already in use
! at org.eclipse.jetty.setuid.SetUIDListener.lifeCycleStarting(SetUIDListener.java:213) ~[jetty-setuid-java-1.0.3.jar:na]
! at org.eclipse.jetty.util.component.AbstractLifeCycle.setStarting(AbstractLifeCycle.java:188) ~[jetty-util-9.2.14.v20151106.jar:9.2.14.v20151106]
! at org.eclipse.jetty.util.component.AbstractLifeCycle.start(AbstractLifeCycle.java:67) ~[jetty-util-9.2.14.v20151106.jar:9.2.14.v20151106]
! at io.dropwizard.cli.ServerCommand.run(ServerCommand.java:43) ~[dropwizard-core-0.9.2.jar:0.9.2]
! at io.dropwizard.cli.EnvironmentCommand.run(EnvironmentCommand.java:41) [dropwizard-core-0.9.2.jar:0.9.2]
! at io.dropwizard.cli.ConfiguredCommand.run(ConfiguredCommand.java:77) [dropwizard-core-0.9.2.jar:0.9.2]
! at io.dropwizard.cli.Cli.run(Cli.java:70) [dropwizard-core-0.9.2.jar:0.9.2]
! at io.dropwizard.Application.run(Application.java:80) [dropwizard-core-0.9.2.jar:0.9.2]
! at com.mesosphere.dcos.cassandra.scheduler.Main.main(Main.java:27) [cassandra-scheduler.jar:na]
```
but the framework process does not exit without this change.

File: cassandra-scheduler/src/main/java/com/mesosphere/dcos/cassandra/scheduler/Main.java [hunk 1]
```diff
@@ -20,9 +20,15 @@
...
   private static final Logger LOGGER = LoggerFactory.getLogger(Main.class);
+  private static final int EXIT_CODE = 7;
 
   public static void main(String[] args) throws Exception {
-    new Main().run(args);
+    try {
+      new Main().run(args);
+    } catch (Exception e) {
+      LOGGER.error("Caught exception while trying to run main: exiting framework process with code: " + EXIT_CODE);
+      System.exit(EXIT_CODE);
```
Comment (mgummelt) [inline cassandra-scheduler/src/main/java/com/mesosphere/dcos/cassandra/scheduler/Main.java]: Let's use this: https://github.com/mesosphere/dcos-commons/blob/master/sdk/scheduler/src/main/java/com/mesosphere/sdk/scheduler/SchedulerErrorCode.java#L6

Comment (verma7) [inline cassandra-scheduler/src/main/java/com/mesosphere/dcos/cassandra/scheduler/Main.java]: @mgummelt Should I use INITIALIZATION_FAILURE or ERROR do you want me to add another code there?

Comment (mgummelt) [inline cassandra-scheduler/src/main/java/com/mesosphere/dcos/cassandra/scheduler/Main.java]: I think ERROR is fine since this is a blanket Exception

Comment (verma7) [inline cassandra-scheduler/src/main/java/com/mesosphere/dcos/cassandra/scheduler/Main.java]: Using ERROR.

**Comment by verma7** [comment]
Done.

**Comment by mgummelt** [comment]
LGTM.  Good to merge after tests pass.

**Comment by verma7** [comment]
@mgummelt Can you let me know why the build is failing? It's building fine on my machine and running fine my vagrant environment.

**Comment by mgummelt** [comment]
retest this please

**Comment by verma7** [comment]
retest this please

**Comment by verma7** [comment]
retest this please

**Comment by verma7** [comment]
retest this please

**Comment by verma7** [comment]
@mgummelt @gabrielhartmann All the tests have passed. Can you please merge this PR?

**Comment by mgummelt** [comment]
Thanks!

**Pull request merged**